src/operator/pooling-inl.h

```diff
@@ -57,13 +57,13 @@ struct PoolingParam : public dmlc::Parameter<PoolingParam> {
               "kValid is default setting of Mxnet and rounds down the output pooling size."
               "kFull is compatible with Caffe and rounds up the output pooling size.");
 
-    int stride_shape[] = {1, 1};
-    DMLC_DECLARE_FIELD(stride).set_default(TShape(stride_shape, stride_shape + 2))
+    int stride_shape[] = {1, 1, 1};
+    DMLC_DECLARE_FIELD(stride).set_default(TShape(stride_shape, stride_shape + kernel.ndim()))
```

Comment:
Set default to TShape() here and set it to 1, 1 or 1, 1, 1 depending on kernel dim in infershape